### ISSUE TITLE
Do not want to clip values into (0,1) for some applications. 

### ISSUE THREAD

**Issue by apple2373** [haruishi43/equilib#13]
I was trying to use this for depth prediction, but I found it problematic that values will be forced into (0,1).

E.g.
https://github.com/haruishi43/equilib/blob/de35bf91685535285bcbea91332bf6710db01643/equilib/equi2cube/torch.py#L238

There's many of them (we can search the keyword of `clip`) and currently there's no way to disable this clipping except changing the source code directly.

**Comment by apple2373** [comment]
Well, it may not be simply removing the torch.clip operations. I don't have motivation to check the implementations carefully, but it seems that this library do assume that tensor values are in  0.0-1.0, so cannot be used for float tensors whose values at its current form. 

https://github.com/haruishi43/equilib/blob/71a88af6545fa5fbb0d2adc030f726de0243205a/equilib/equi2cube/torch.py#L161-L162

**Comment by haruishi43** [comment]
@apple2373 does the function work correctly for depth data? is it int32 or float?

My purpose for this library was RGB images so if you would like to add depth image functionality, I would appreciate it.
Otherwise, I'm not too familiar with panoramic depth data so if you could give me some sample data, I might give it a try later.

**Comment by apple2373** [comment]
No, this library doesn't work correctly for depth data at its current form. The input data is float, and its values are not limited to the range from 0 to 1. 

I haven't decided my mind to extend this library, find another ready-to-use one, or even implement it myself. If I happen to choose the first option, i'll update here. 

I don't have data to share but you could check popular benchmarks like 2D3D dataset ( https://github.com/alexsax/2D-3D-Semantics ), which has panoramic depths if you are really motivated to work on panoramic depth prediction. 

But the issue I wanted to raise ware nothing specific to panoramic depth. This library is currently not compatible with float tensors that can range outside of 0-1. But given that this library is for RGB images, as you state, then maybe that's not an issue.

**Comment by haruishi43** [comment]
@apple2373 I was asking if the function works when you remove the `clip(0,1)`. You said there is a lot of `clip`, but there is only one `clip` if you're planning to use `equi2cube.torch.run` (I think you were looking at this right?):
```
out = (
    out.type(equi_dtype)
    if equi_dtype == torch.uint8
    else torch.clip(out, 0.0, 1.0)
)
```
It would work if these lines were commented out. I needed to add this to ensure that return formats were correct since float and uint8 are allowed.
If you're already using this function, I would love to see if it works or not.

Thanks for the insight! I'll try to support depth formats as well in the future, but currently, I dont' have much time to test this on my own.

**Comment by apple2373** [comment]
Unfortunately i couldn't conclude if removing torch.clip is enough to be compatible with float tensor or not. I indeed tried removing torch.clip, but I can't confidently say if there's no other problem or not. What I tried is to transform a equirectangular depth map into a cube map by equi2cube, and then reverse it back by cube2equi. They seems to be visually similar, but they are not identical. Then I stopped investigating more.
